### PR TITLE
WIP: Bug 1070716 - Only allow Admin/Council to change mentor

### PR DESCRIPTION
**Pull request by MichaelKohler**
https://bugzilla.mozilla.org/show_bug.cgi?id=1070716
This is a work in progress.

The tests pass, however it doesn't work when I test this with a Council account. The field doesn't show up since the perm doesn't get applied to the council member. How can I make sure that existing users get that permission as well?

@akatsoulas not important right now, let's focus on the other PRs for now. However, once you have time, would love to learn more about this :)

### CODE REVIEW

**Comment by MichaelKohler** [comment]
@akatsoulas a small hint would be appreciated ;)

**Comment by akatsoulas** [comment]
@MichaelKohler you need to update the `clean` method of the form [0]. Basically, the clean method checks the group that the user is member of and decides whether the field is required or not.

[0] https://github.com/mozilla/remo/blob/master/remo/profiles/forms.py#L144

**Comment by MichaelKohler** [comment]
This works now, thanks for the help. Could you please review again?

File: remo/profiles/apps.py [hunk 1]
```diff
@@ -11,8 +11,9 @@ def profiles_set_groups(sender, **kwargs):
         return True
 
     perms = {'create_user': ['Admin', 'Mentor'],
-             'can_edit_profiles': ['Admin'],
-             'can_delete_profiles': ['Admin']}
+             'can_edit_profiles': ['Admin', 'Council', 'Peers'],
```
Comment (akatsoulas) [inline remo/profiles/apps.py]: Let's drop Peers from here. The less people have edit access here the better.

File: remo/profiles/forms.py [hunk 2]
```diff
@@ -143,8 +143,9 @@ def clean(self):
 
         # Do not raise a validation error if the user belongs to the Alumni
         # group or has admin privileges
-        if (not mentor and not
-                self.request.user.groups.filter(name='Admin').exists()):
+        if (not mentor and
```
Comment (akatsoulas) [inline remo/profiles/forms.py]: this probably should be `and not`.

File: remo/profiles/templates/profiles_edit.jinja [hunk 3]
```diff
@@ -202,14 +202,16 @@
...
             {% endif %}
-            <div class="has-tip tip-left" title="Mentor" data-tooltip data-options="disable-for-touch:true">
-              {{ profileform.mentor }}
-            </div>
-            <div class="required-field"></div>
-            {% if profileform.mentor.errors %}
-              <small class="error">
-                {{ profileform.mentor.errors }}
-              </small>
+            {% if perms.profiles.can_change_mentor %}
```
Comment (akatsoulas) [inline remo/profiles/templates/profiles_edit.jinja]: That's nice!

File: remo/profiles/apps.py (hunk 1 above)
Comment (MichaelKohler) [reply remo/profiles/apps.py]: You're right. Will do.

File: remo/profiles/forms.py (hunk 2 above)
Comment (MichaelKohler) [reply remo/profiles/forms.py]: Thinking about this, we probably can remove that whole block. As anybody else won't be able to change it, and for Admin and Council it indeed should be optional, we don't need it. Do you agree?

**Comment by MichaelKohler** [comment]
@akatsoulas did the requested changes, r?

File: remo/profiles/migrations/0009_auto_20171221_0112.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
```
Comment (akatsoulas) [inline remo/profiles/migrations/0009_auto_20171221_0112.py]: There is already a migration in the profiles app numbered as `0009`.

File: remo/profiles/tests/test_models.py [hunk 5]
```diff
@@ -281,19 +281,23 @@ def setUp(self):
...
             ok_(admin.has_perm(permission))
 
-    def test_council_group_has_no_permissions(self):
-        """Test that council group has no permissions."""
+    def test_council_group_has_mentor_change_permissions(self):
+        """Test that council group has mentor change permissions."""
         councelor = UserFactory.create(groups=['Council'])
         for permission in self.permissions:
-            ok_(not councelor.has_perm(permission))
+            if permission == 'profiles.can_change_mentor':
```
Comment (akatsoulas) [inline remo/profiles/tests/test_models.py]: I would avoid having an `if` in a test. Probably this means that you can break this test into two and test both cases.

**Comment by MichaelKohler** [comment]
@akatsoulas did the requested changes again, looking forward to your final review :)

**Comment by MichaelKohler** [comment]
@akatsoulas rebased to master

**Pull request merged**